src/MergeContext.js

```diff
@@ -405,17 +474,16 @@ class MergeContext {
         const ret = (prevLen === requiredChecks.length);
         if (ret && andSupplyRequired) {
             for (let requiredContext of requiredContexts) {
-                // a passed check misses a required one
+                // go further only if the passed check context is not a required one
                 if (requiredChecks.find(el => el.context.trim() === requiredContext.trim()) === undefined) {
-                    // but the passed check matches a required one
+                    // go further only if the passed check context matches the required one
```

Comment:
```
// go further only if the passed check context is not a required one 
// go further only if the passed check context matches the required one 
```

It looks like we are now implying that a "passed check context" may "not be a required one" but may "match the required one", which is still very confusing because we should not even bother with non-required checks. The whole intent of this code is unclear and the comments are not helping. Perhaps there is a wider terminology/naming problem. Perhaps the overall intent should be documented _before_ the loop, using consistent terminology. Unfortunately, I do not understand this code enough to suggest a specific fix.

I cannot spend more time on this mess. You can ignore this complaint if you are sure the code "works".